crates/compiler/gen_llvm/src/llvm/build.rs

```diff
@@ -5978,23 +5876,11 @@ fn build_closure_caller<'a, 'ctx>(
 ) {
     let mut argument_types = Vec::with_capacity_in(arguments.len() + 3, env.arena);
 
-    for layout in arguments {
-        let arg_type =
-            basic_type_from_layout(env, layout_interner, layout_interner.get_repr(*layout));
-        let arg_ptr_type = arg_type.ptr_type(AddressSpace::default());
-
-        argument_types.push(arg_ptr_type.into());
+    for _ in arguments {
+        argument_types.push(env.context.ptr_type(AddressSpace::default()).into());
```

Comment:
nitpick: we could change this to something like the below for clarity:

```rust
let ptr_type = env.context.ptr_type(AddressSpace::default());
argument_types.extend(std::iter::repeat(ptr_type).take(arguments.len()));
```

Reply:
I find the current way is clearer... but I'm less familiar with extend, repeat and take.

Reply:
super minor, I'll just resolve.